markets/storageadapter/dealpublisher.go

```diff
@@ -257,11 +279,13 @@ func (p *DealPublisher) waitForMoreDeals() {
 }
 
 func (p *DealPublisher) publishAllDeals() {
-	// If the timeout hasn't yet been cancelled, cancel it
-	if p.cancelWaitForMoreDeals != nil {
-		p.cancelWaitForMoreDeals()
-		p.cancelWaitForMoreDeals = nil
-		p.publishPeriodStart = time.Time{}
+	if !p.manualPSD {
```

Comment:
I don't think you need to check against p.manualPSD here - as long as we never call waitForMoreDeals, cancelWaitForMoreDeals will always be nil